### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by generatedman — APPROVED** [review]
good job!

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,14 @@
...
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(man => (
+    century
+      ? man.sex === 'm' && Math.ceil(man.died / 100) === century
+      : man.sex === 'm'
+  ));
+  const menAges = men.map(man => man.died - man.born);
```
Comment (silchyn) [inline src/averageAges.js:23]:
```suggestion
  const men = people.filter(man => (
    man.sex === 'm' && (century
      ? && Math.ceil(man.died / 100) === century
      : true)
  ));
  
  const menAges = men.map(man => man.died - man.born);
```

File: src/averageAges.js [hunk 2]
```diff
@@ -55,7 +65,21 @@ function calculateWomenAverageAge(people, withChildren) {
...
+  ));
+  const ageDifferences = children.map(child => (
+    child.born - people.find(mother => mother.name === child.mother).born
+  ));
+
+  return getAverageAge(ageDifferences, children.length);
+}
+
+function getAverageAge(ages, amount) {
+  return ages.reduce((sum, age) => sum + age, 0) / amount;
```
Comment (silchyn) [inline src/averageAges.js:82]: what if amount is 0?

File: src/averageAges.js [hunk 3]
```diff
@@ -55,7 +65,21 @@ function calculateWomenAverageAge(people, withChildren) {
...
-  // write code here
+  const children = people.filter(child => (
+    onlyWithSon
+      ? people.find(mother => mother.name === child.mother)
+        && child.sex === 'm'
+      : people.find(mother => mother.name === child.mother)
+  ));
+  const ageDifferences = children.map(child => (
+    child.born - people.find(mother => mother.name === child.mother).born
+  ));
```
Comment (silchyn) [inline src/averageAges.js:76]:
```suggestion
  const children = people.filter(child => (
    onlyWithSon
      ? people.find(mother => mother.name === child.mother)
        && child.sex === 'm'
      : people.find(mother => mother.name === child.mother)
  ));
  
  const ageDifferences = children.map(child => {
    const mother = people.find(mother => mother.name === child.mother);
  
    return child.born - mother.born;
  );
```

**Review by silchyn — CHANGES_REQUESTED** [review]

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**